### ISSUE TITLE
Angular Reactive forms custom form controls inconsistency in validations after conditionally recreated by angular

### ISSUE THREAD

**Issue by sankarswaroop** [angular/angular#26792]
## I'm submitting a...

<pre><code>
[x] Bug report 
</code></pre>

## Current behavior
Reactive-Forms:
Custom Form Control validations stick to initial state when recreating this control conditionally by angular

## Expected behavior
The validations should work the same every time this control is recreated

## Minimal reproduction of the problem with instructions
https://stackblitz.com/edit/angular-4ansfk
or
https://angular-4ansfk.stackblitz.io
1. Load the above url. 
2. After the screen loaded, click 'toggle' first and you can see 'Custom Name' field.
3. Type in text input and the error goes off.
4. Clear text and you can see error is back
5. Now click 'toggle' and you can see 'First Name' field
6. Click 'toggle' and you can see 'Custom Name' field with error message
7. Now type in the textinput and you can still see the error shows. The error message should go off as per the validations i written.

## What is the motivation / use case for changing the behavior?
Consistent validations with custom form controls while recreating by angular conditionally

## Environment

<pre><code>
Angular version: 5.2.10
I cannot upgrade to latest version and I have observed the same in 4.6.4 too

Browser:
- [x] Chrome (desktop) version 70.0 or earlier too
- [x] IE version 11
 
For Tooling issues:
- Node version: 8.2.0
- Platform:  Windows
</code></pre>

**Comment by lloydst** [comment]
you forgot to update the dom, add a update rule:
```
constructor(private fb: FormBuilder) {
    this.formGroup = this.fb.group({
      firstName: new FormControl(),
      customName: new FormControl('', [Validators.required]),
    }, {updateOn:blur});
    this.toggle();
```

**Comment by sankarswaroop** [comment]
@lloydst Thanks for your replay. This is more about on what action the validations should run. This will execute validations on blur of each control. This doesn't solved the actual problem

**Comment by AndrewKushnir** [comment]
Hi @sankarswaroop, thanks for reporting the issue. It looks related to #20007 and #34538. I'll keep this ticket open, so this use-case can be verified once the issue is fixed (no ETA at this moment). Thank you.

**Comment by AndrewKushnir** [comment]
Closing this issue since it has the same root cause as #20007 and should be resolved by #39235.
